server/src/test/java/org/apache/iotdb/db/metadata/mtree/schemafile/RecordUtilTests.java

```diff
@@ -54,15 +56,22 @@ public void tearDown() throws Exception {
 
   @Test
   public void internalNodeTest() throws MetadataException {
-    IMNode oneNode = new InternalMNode(null, "abcd");
+    ICacheMNode oneNode = nodeFactory.createInternalMNode(null, "abcd");
+    ICacheMNode twoNode = nodeFactory.createDeviceMNode(null, "efgh").getAsMNode();
     ICachedMNodeContainer.getCachedMNodeContainer(oneNode).setSegmentAddress(1234567L);
-    oneNode.setUseTemplate(true);
+    ICachedMNodeContainer.getCachedMNodeContainer(twoNode).setSegmentAddress(66666L);
+    twoNode.getAsDeviceMNode().setUseTemplate(true);
     ByteBuffer buffer = RecordUtils.node2Buffer(oneNode);
     buffer.clear();
-    IMNode node2 = RecordUtils.buffer2Node("abcd", buffer);
+    ICacheMNode node1 = RecordUtils.buffer2Node("abcd", buffer);
+    Assert.assertEquals(
+        ICachedMNodeContainer.getCachedMNodeContainer(node1).getSegmentAddress(), 1234567L);
```

Comment:
The true value is supposed to be placed as 1st param

Reply:
Fixed~